shared/common-adapters/avatar.desktop.js

```diff
@@ -4,67 +4,80 @@ import React, {PureComponent} from 'react'
 import {globalStyles, globalColors} from '../styles'
 import {resolveImageAsURL} from '../../desktop/resolve-root'
 
-import type {Props, AvatarLookup} from './avatar'
+import type {Props, AvatarLookup, AvatarLoad} from './avatar'
 
 const noAvatar = resolveImageAsURL('icons', 'icon-placeholder-avatar-112-x-112@2x.png')
 // To convert usernames/uids to avatarurls
 let _urlLookup
+let _urlLoad
 
 type State = {
   avatarLoaded: boolean,
   errored: boolean,
+  url: ?string,
 }
 
 class Avatar extends PureComponent<void, Props, State> {
   state: State;
-  _onURLLoaded: ?() => void;
+  _onURLLoaded: ?(username: string, url: ?string) => void;
 
   constructor (props: Props) {
     super(props)
-    this.state = {avatarLoaded: false, errored: false}
+
+    const urlState = this._getUrlState(props)
+    this.state = {avatarLoaded: false, errored: false, ...urlState}
   }
 
   componentDidMount () {
-    this._onURLLoaded = () => { this.forceUpdate() }
+    this._onURLLoaded = (username: string, url: ?string) => {
+      if (this.props.username === username) {
+        this.setState({url})
```

Comment:
mmmm, i think its not valid for the setState to be called outside of the mount/unMount so i think we should leave this as-is